### ISSUE TITLE
BTrace causes JVM to crash on call tracing

### ISSUE THREAD

**Issue by Kshitiz-Sharma** [btraceio/btrace#240]
I have following trace script:

```
@BTrace
public class MethodCallTracer {
    @OnMethod(clazz = "/.*/", method = "/.*/",
            location = @Location(value = Kind.CALL, clazz = "/.*/", method = "myMethod"))
    public static void m(@Self Object self, @TargetMethodOrField String method,
                         @ProbeMethodName String probeMethod) {
        println(method + " in " + probeMethod);
    }
}
```

That I'm running with following program:

```
public class MethodCallTracingExample {
    public static void main(String[] args) throws InterruptedException {
        while (true) {
            Thread.sleep(2000);
        }
    }
    public int myMethod() {
        return -7;
    }
}
```

It causes the JVM to crash:

```
"C:\Program Files\Java\jdk1.8.0_92\bin\java" -Didea.launcher.port=7533 "-Didea.launcher.bin.path=C:\Program Files (x86)\JetBrains\IntelliJ IDEA Community Edition 2016.1.1\bin" -Dfile.encoding=UTF-8 -classpath "C:\Program Files\Java\jdk1.8.0_92\jre\lib\charsets.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\deploy.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\access-bridge-64.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\cldrdata.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\dnsns.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\jaccess.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\jfxrt.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\localedata.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\nashorn.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\sunec.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\sunjce_provider.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\sunmscapi.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\sunpkcs11.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\ext\zipfs.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\javaws.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\jce.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\jfr.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\jfxswt.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\jsse.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\management-agent.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\plugin.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\resources.jar;C:\Program Files\Java\jdk1.8.0_92\jre\lib\rt.jar;C:\Users\kshar\Documents\OneDrive\Revenue Integrity\Sprints\1608\JPDA\BTrace-Example\target\test-classes;C:\Users\kshar\Documents\OneDrive\Revenue Integrity\Sprints\1608\JPDA\BTrace-Example\target\classes;C:\Development\Software\btrace-1.3.8\build\btrace-boot.jar;C:\Program Files (x86)\JetBrains\IntelliJ IDEA Community Edition 2016.1.1\lib\idea_rt.jar" com.intellij.rt.execution.application.AppMain MethodCallTracingExample
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x0000000057c2a37a, pid=4572, tid=0x0000000000001e90
#
# JRE version: Java(TM) SE Runtime Environment (8.0_92-b14) (build 1.8.0_92-b14)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.92-b14 mixed mode windows-amd64 compressed oops)
# Problematic frame:
# V  [jvm.dll+0x13a37a]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# C:\Users\kshar\Documents\OneDrive\Revenue Integrity\Sprints\1608\JPDA\BTrace-Example\hs_err_pid4572.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
Process finished with exit code 1

```

**Comment by Kshitiz-Sharma** [comment]
Doesn't seem to be a bug but wrong use of annotations. Fixes itself when more specific pointcut expression is added:
```
@OnMethod(clazz = "/.*MethodCallTracingExample.*/", method = "/.*/",
            location = @Location(value = Kind.CALL, clazz = "/.*/", method = "myMethod"))
```
I'm not familiar with the code but I guess the crash could be due to BTrace trying to weave in the advice in all the classes. 

The fix for this could be perhaps adding a check into the script compiler to ensure that expression isn't too generic and emit a warning.

Although I'll leave the issue open for now Jaroslav's comments.

**Comment by szhnet** [comment]
I found the same problem on jdk 1.8, but jdk 1.7 no problem.

**Comment by jbachorik** [comment]
Yes, there are certain classes in JDK which don't like being instrumented. We keep a list of known classes causing problems when instrumented and BTrace is skipping them. Apparently, there have been added some more classes that should be excluded.

**Comment by jbachorik** [comment]
Has been resolved in #256
